infoview/suggestions.tsx

```diff
@@ -0,0 +1,157 @@
+    }, [props.reqId]);
+
+    return (
+        <div>
+            <p>
+                <span className={'easeTransition ml2 b-ns ' + statusColTable[status]}>Suggestions</span>
+            </p>
+            <div>
+                {['loading', 'error'].includes(status) ?
+                    <p className={'easeTransition ml3 ' + statusColTable[status]}>
+                        {status === 'loading' ? 'Querying Suggestions' : 'Something went wrong :('}
+                        {status === 'loading'? <Waiter />: null}
+                    </p>
+                : null}
+                {errorMsg ? <i className={'easeTransition ml3 ' + statusColTable[status]}>{errorMsg}</i> : null}
+                {(['done', 'checking'].includes(status) && suggestionsResults && !suggestionsResults.error && suggestionsResults.tactic_infos) ?
+                    <div>{Object.values(suggestionsResults.tactic_infos).map(
+                        tactic_info => <div><SingleTacticInfo key={tactic_info.tactic} tacticInfo={tactic_info} widget={props.widget}/><br/></div>
+                    )}</div>
+                    : null
+                }
+            </div>
+        </div>
+    );
+}
+
+export function Suggestor(props: SuggestionsProps): JSX.Element {
+    const [suggReqId, setSuggReqId] = React.useState(0);
+    const [doSuggest, setDoSuggest] = React.useState(false);
+    const [errorMsg, setErrorMsg] = React.useState<string>();
+    const [prefix, setPrefix] = React.useState('');
+
+    useEvent(
+        SuggestionsErrorEvent,
+        (s) => {
+            setErrorMsg(s.error);
+        }, []
+    );
+
+    React.useEffect(() => {
+        setDoSuggest(true);
+    }, [prefix, props.goalState])
+    React.useEffect(delayedThrottled(
+        300,
+        () => {
+            if (doSuggest) setDoSuggest(false);
+            else return;
+            if (props.goalState !== undefined && props.goalState !== 'no goals') {
+                setSuggReqId(rid => {
+                    post({
+                        command: 'get_suggestions',
+                        reqId: rid + 1,
+                        goalState: props.goalState,
+                        prefix
+                    });
+                    setErrorMsg(undefined); // reset error msg
+                    return rid + 1
+                });
+            }
+        }
+    ), [prefix, props.goalState, doSuggest])
+
+    const inputProps = { style: { display: 'inline' }, type: 'text', size: 12, value: prefix };
+
+    return (
+        <div>
+            <details open={true}>
+                <summary className={'mv2'}>
+                    Suggestions
+                </summary>
+                {errorMsg
+                    ? <i className={'easeTransition mv2 ' + statusColTable.error}>{errorMsg}</i>
+                    : <div>
+                        <form className='ml2' onSubmit={(e) => e.preventDefault()}>
+                            <span>Tactic prefix: </span>
```

Comment:
```suggestion
                            <span>Tactic suggestions with prefix: </span>
```

The double collapsing (with both the button and the details) is a bit weird.